modules/dbnd-airflow/src/dbnd_airflow/scheduler/single_dag_run_job.py

```diff
@@ -729,6 +728,16 @@ def _execute(self, session=None):
                 executor.end()
             except Exception:
                 logger.exception("Failed to terminate executor")
+            if dag_run.state == State.RUNNING:
+                TI = TaskInstance
+                session.query(TI).filter(
+                    TI.dag_id == self.dag_id, TI.execution_date == self.execution_date,
```

Comment:
Same here it would make sense to change state only of TIs that have `state != State.SUCCESS`